proto/connect/agent_service/v1alpha1/agent_service.proto

```diff
@@ -5,32 +5,44 @@ syntax = "proto3";
 
 package proto.connect.agent_service.v1alpha1;
 
+import "proto/agent/v1alpha1/agent.proto";
 import "proto/federated_service/v1alpha1/federated_service.proto";
 import "proto/spire/api/types/bundle.proto";
 
-option go_package = "github.com/cofide/cofide-api-sdk/gen/go/proto/agent_service/v1alpha1";
+option go_package = "github.com/cofide/cofide-api-sdk/gen/go/proto/connect/agent_service/v1alpha1";
 
 service AgentService {
+  rpc CreateAgentJoinToken(CreateAgentJoinTokenRequest) returns (CreateAgentJoinTokenResponse) {}
+  rpc AuthenticateAgent(AuthenticateAgentRequest) returns (AuthenticateAgentResponse) {}
```

Comment:
Still unsure about the naming of this one. It performs two functions:

1. Register the agent using a join token, and get an agent ID in response. This creates the agent entry in the connect DB.
2. Bootstrap federation with connect by uploading a bundle.

Potentially the join token could go away in a TZaaS environment, but it still might be useful to have an agent registration process? In that case the join token could be optional for an agent with a SPIFFE ID?

Reply:
RegisterAgent does feel more encompassing of the actions you outlined:

> 
 1.  Register the agent using a join token, and get an agent ID in response. This creates the agent entry in the connect DB.
 2. Bootstrap federation with connect by uploading a bundle.

I do like the idea of streamlining this process to remove the join token!

Reply:
Decided to leave RegisterAgent for now.